File: php/129-sum-root-to-leaf-numbers.php
<?php
/**
 * Definition for a binary tree node.
 * class TreeNode {
 *     public $val = null;
 *     public $left = null;
 *     public $right = null;
 *     public function __construct($value) { $this->val = $value; }
 * }
 */
class Solution 
{
    private $sum = 0;
    /**
     * @param TreeNode $root
     * @return Integer
     */
    public function sumNumbers($root) 
    {
        if ($root == null) {
            return 0;
        }
        $this->DFS($root);
        return $this->sum;
    }
    public function DFS($node, $path = []) 
    {
        if ($node == null) {
            return ;
        }
        $path[] = $node->val;
        if ($node->left == null && $node->right == null) {
            $this->sum += intval(implode("", $path));
            return ;
        }
        $this->DFS($node->left, $path);
        $this->DFS($node->right, $path);
    }
}

File: go/129-sum-root-to-leaf-numbers.go
package main

/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
func sumNumbers(root *TreeNode) int {
	if root == nil {
		return 0
	}
	sum := 0
	sumNodes(root, []int{}, &sum)
	return sum

}
func sumNodes(root *TreeNode, num []int, sum *int) {
	if root == nil {
		return
	}
	if root.Left == nil && root.Right == nil {
		*sum += toNumber(append(num, root.Val))
		return
	}
	numCopy := make([]int, len(num))
	copy(numCopy, num)
	numCopy = append(numCopy, root.Val)
	if root.Left != nil {
		sumNodes(root.Left, numCopy, sum)
	}
	if root.Right != nil {
		sumNodes(root.Right, numCopy, sum)
	}
}
func toNumber(num []int) int {
	base := 1
	number := 0
	for i := len(num) - 1; i >= 0; i-- {
		number += base * num[i]
		base *= 10
	}
	return number
}
